app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -918,47 +917,93 @@ else if(needed == SessionFrame.STATE_COMMAND_ID) {
-        try {
-            //If this is a new record (never saved before), which currently all should be 
-            if(state.getFormRecordId() == -1) {
-                    
-                //If form management isn't enabled we can't have these old forms around anyway
-                if(CommCarePreferences.isIncompleteFormsEnabled()) {
-                    //First, see if we've already started this form before
-                    SessionStateDescriptor existing = state.searchForDuplicates();
-                    
-                    //I'm not proud of the second clause, here. Basically, only ask if we should continue entry if the
-                    //saved state actually involved selecting some data.
-                    if(existing != null && existing.getSessionDescriptor().contains(SessionFrame.STATE_DATUM_VAL)) {
-                        createAskUseOldDialog(state, existing);
-                        return;
-                    }
+        if (state.getFormRecordId() == -1) {
+            if (CommCarePreferences.isIncompleteFormsEnabled() &&
+                    state.getSessionStateDescriptor().getSessionDescriptor().contains(SessionFrame.STATE_DATUM_VAL)) {
+                // Are existing (incomplete) forms using the same case?
+                SessionStateDescriptor existing = state.searchForDuplicates();
+                if (existing != null) {
+                    // Ask user if they want to just edit existing form
+                    // that uses the same case.
+                    createAskUseOldDialog(state, existing);
+                    return;
                 }
-                
-                //Otherwise, generate a stub record and commit it
-                state.commitStub();
-            } else {
-                Logger.log("form-entry", "Somehow ended up starting form entry with old state?");
             }
-            
-            //We should now have a valid record for our state. Time to get to form entry.
-            FormRecord record = state.getFormRecord();
 
-            if (platform == null &&
-                    CommCareApplication._().getCurrentApp() != null) {
-                platform = CommCareApplication._().getCommCarePlatform();
-            }
+            // Generate a stub form record and commit it
+            state.commitStub();
+        } else {
+            Logger.log("form-entry",
+                    "Somehow ended up starting form entry with old state?");
+        }
 
-            //TODO: May need to pass session over manually
-            formEntry(platform.getFormContentUri(record.getFormNamespace()), record, CommCareActivity.getTitle(this, null));
-            
-        } catch (StorageFullException e) {
-            throw new RuntimeException(e);
+        FormRecord record = state.getFormRecord();
+
+        if (platform == null &&
+                CommCareApplication._().getCurrentApp() != null) {
+            platform = CommCareApplication._().getCommCarePlatform();
         }
```

Comment:
Removed this try/catch because StorageFullException extends RuntimeException, so why catch and re-raise?
I can add it back in if you see importance to this.